common/test/acceptance/pages/lms/courseware.py

```diff
@@ -319,3 +328,258 @@ def get_selected_tab_content(self):
+    def is_on_section(self, section_title, subsection_title):
+        """
+        Return a boolean indicating whether the user is on the section and subsection
+        with the specified titles.
+
+        """
+        # TODO: TNL-6546: Remove if/else; always use unified_course_view version (if)
+        if self.unified_course_view:
+            # breadcrumb location of form: "SECTION_TITLE > SUBSECTION_TITLE > SEQUENTIAL_TITLE"
+            bread_crumb_current = self.q(css='.position').text
+            if len(bread_crumb_current) != 1:
+                self.warning("Could not find the current bread crumb with section and subsection.")
+                return False
+
+            return bread_crumb_current[0].strip().startswith(section_title + ' > ' + subsection_title + ' > ')
+
+        else:
+            # This assumes that the currently expanded section is the one we're on
+            # That's true right after we click the section/subsection, but not true in general
+            # (the user could go to a section, then expand another tab).
+            current_section_list = self.q(css='.course-navigation .chapter.is-open .group-heading').text
+            current_subsection_list = self.q(css='.course-navigation .chapter-content-container .menu-item.active a p').text
+
+            if len(current_section_list) == 0:
+                self.warning("Could not find the current section")
+                return False
+
+            elif len(current_subsection_list) == 0:
+                self.warning("Could not find current subsection")
+                return False
+
+            else:
+                return (
+                    current_section_list[0].strip() == section_title and
+                    current_subsection_list[0].strip().split('\n')[0] == subsection_title
+                )
+
+    # Regular expression to remove HTML span tags from a string
+    REMOVE_SPAN_TAG_RE = re.compile(r'</span>(.+)<span')
+
+    def _clean_seq_titles(self, element):
+        """
+        Clean HTML of sequence titles, stripping out span tags and returning the first line.
+        """
+        return self.REMOVE_SPAN_TAG_RE.search(element.get_attribute('innerHTML')).groups()[0].strip()
+
+    # TODO: TNL-6546: Remove from here and move to course_home.py:CourseOutlinePage
+    @property
+    def active_subsection_url(self):
+        """
+        return the url of the active subsection in the left nav
+        """
+        return self.q(css='.chapter-content-container .menu-item.active a').attrs('href')[0]
+
+    # TODO: TNL-6546: Remove all references to self.unified_course_view
+    # TODO: TNL-6546: Remove the following function
+    def visit_unified_course_view(self):
+        # use unified_course_view version of the nav
+        self.unified_course_view = True
+        # reload the same page with the unified course view
+        self.warning("CourseNavPage.visit_unified_course_view: before browser get: url='{0}'".format(self.browser.current_url))
```

Comment:
@andy-armstrong Once my test is passing, I need to remove a commit that added this additional logging.  I'm just leaving it in for now in case my fix didn't work.